Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool+Waiter.swift

```diff
@@ -0,0 +1,64 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2021 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+extension HTTPConnectionPool {
+    struct RequestID: Hashable {
+        private let objectIdentifier: ObjectIdentifier
+
+        init(_ request: HTTPScheduledRequest) {
+            self.objectIdentifier = ObjectIdentifier(request)
+        }
+    }
+
+    struct Waiter {
+        var requestID: RequestID {
+            RequestID(self.request)
+        }
+
+        var request: HTTPScheduledRequest {
+            didSet {
```

Comment:
Might it be easier to make `eventLoopRequirement` a computed property?

Reply:
Since `HTTPScheduledRequest` is a protocol, I wanted to keep the number of calls going through the lookup table as small as possible.